hal/lib/athena/Counter.cpp

```diff
@@ -365,7 +365,8 @@ double HAL_GetCounterPeriod(HAL_CounterHandle counter_handle, int32_t* status) {
     period = static_cast<double>(output.Period << 1) /
              static_cast<double>(output.Count);
   }
-  return period * 2.5e-8;  // result * timebase (currently 40ns)
+  return static_cast<float>(period *
```

Comment:
Is this well-defined in the infinity case?  Inf * something, cast to float..  And actually, I'm not sure it's the best idea to divide by zero to try to get infinity (if you're lucky, you'll get NAN, but not infinity)... seems risky, we should fix it to actually use the proper constant.